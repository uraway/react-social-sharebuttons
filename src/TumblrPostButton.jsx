import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class TumblrPostButton extends Component {
  constructor(props) {
    super(props);
    this.state = ({ initialized: false });
  }

  componentDidMount() {
    if (this.state.initialized) {
      return;
    }

    const tumblrbutton = this.node;
    const tumblrscript = document.createElement('script');
    tumblrscript.src = 'https://secure.assets.tumblr.com/share-button.js';
    tumblrscript.id = 'tumblr-js';
    tumblrbutton.parentNode.appendChild(tumblrscript);

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    return (
      <a
        ref={node => this.node = node}
        className="tumblr-share-button"
        data-color={this.props.color}
        data-notes={this.props.notes}
        href="https://embed.tumblr.com/share"
      >
        Tumber Post Button
      </a>
    );
  }
}

TumblrPostButton.propTypes = {
  color: PropTypes.string,
  notes: PropTypes.string,
};

TumblrPostButton.defaultProps = {
  color: '',
  notes: ''
};

/*
<a class="tumblr-share-button" data-color="blue" data-notes="right" href="https://embed.tumblr.com/share"></a>
<script>!function(d,s,id){var js,ajs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="https://secure.assets.tumblr.com/share-button.js";ajs.parentNode.insertBefore(js,ajs);}}(document, "script", "tumblr-js");</script>
*/
