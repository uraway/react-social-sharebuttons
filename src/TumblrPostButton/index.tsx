import { FC } from 'react';
import { useLayoutEffectOnce } from '../utils';

export type TumblrPostButtonProps = {
  color?: 'blue' | 'white' | 'black';
  notes?: 'none' | 'top' | 'right';
};

const TumblrPostButton: FC<TumblrPostButtonProps> = (props) => {
  useLayoutEffectOnce(() => {
    const script = document.createElement('script');
    script.src = 'https://secure.assets.tumblr.com/share-button.js';
    script.id = 'tumblr-js';
    document.getElementsByTagName('head')[0]?.appendChild(script);

    return () => script.remove();
  });
  const { color, notes } = props;
  return (
    <div>
      <a className="tumblr-share-button" data-color={color} data-notes={notes} href="https://embed.tumblr.com/share">
        Tumblr Post Button
      </a>
    </div>
  );
};

TumblrPostButton.defaultProps = {
  color: 'blue',
  notes: 'none',
};

export default TumblrPostButton;
