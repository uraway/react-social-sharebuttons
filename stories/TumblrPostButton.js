import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import { TumblrPostButton } from '../dist/index.esm.js';

const colorOptions = {
  blue: 'blue',
  black: 'black',
  white: 'white',
};

const notesOptions = {
  none: 'none',
  top: 'top',
  right: 'right',
};

export default storiesOf('TumblrPostButton', module)
  .addDecorator(withKnobs)
  .add('configured', () => (
    <TumblrPostButton color={select('color', colorOptions)} notes={select('notes', notesOptions)} />
  ));
