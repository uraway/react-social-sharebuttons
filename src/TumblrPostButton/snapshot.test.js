import React from 'react';
import renderer from 'react-test-renderer';
import TumblrPostButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<TumblrPostButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<TumblrPostButton color="black" notes="right" />).toJSON();
  expect(tree).toMatchSnapshot();
});
