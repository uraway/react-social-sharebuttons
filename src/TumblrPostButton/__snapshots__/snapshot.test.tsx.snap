// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="tumblr-share-button"
    data-color="black"
    data-notes="right"
    href="https://embed.tumblr.com/share"
  >
    Tumblr Post Button
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="tumblr-share-button"
    data-color="blue"
    data-notes="none"
    href="https://embed.tumblr.com/share"
  >
    Tumblr Post Button
  </a>
</div>
`;
